scripts/pipeline_generator/step.py

```diff
@@ -0,0 +1,60 @@
+from pydantic import BaseModel, Field
+from typing import List, Dict, Any, Optional
+
+from .utils import AgentQueue
+
+
+class TestStep(BaseModel):
+    """This class represents a test step defined in the test configuration file."""
```

Comment:
why are these in this PR? do you want to test maybe unmarshaling these data structures?

Reply:
the method that unmarshal them from the yaml file would be in the next PR

Reply:
then why is this checked in in this PR? this is not referenced in this PR anywhere?